src/SampleFormat.cpp

```diff
@@ -43,11 +43,10 @@
 #include <string.h>
 
 #include "Prefs.h"
-#include "Dither.h"
 #include "Internat.h"
 
-static DitherType gLowQualityDither = DitherType::none;
-static DitherType gHighQualityDither = DitherType::none;
+DitherType gLowQualityDither = DitherType::none;
```

Comment:
A comment about where these are set to some other value (Preferences) would be helpful.  
A comment that in Audacity there is no default dithering - it is entirely an opt-in feature.

Reply:
See corresponding header file where I made a comment.  I believe header files are better places for such comments.

It is not correct that there is no default dithering.  High quality dither defaults to shaped dither but can be changed in preferences.  It is not statically assigned, but assigned during application initialization, after the config file is open.

See Real-time conversion and High-quality conversion static boxes in the Quality page of Preferences.

Examine constructions of Mixer objects, and you can see high quality dither is used in StereoToMono, export, and score align (but that's commented out).

Otherwise mixers in AudioIO use low quality.

Reply:
The assignment to DitherType::none is misleading.  As '= DitherType::none;' is not in the header, the comment in the header is distant from where it is needed, if it is intended to explain '= DitherType::none;'.

Reply:
That is strictly speaking not assignment but initialization, of a static variable, and cannot be in the header.  It doesn't look good to me to leave variables of nonclass types uninitialized, especially when inside functions, although statics are sure to be initialized with 0.

"Application startup" means something later than static (or here indeed compile-time) initialization time.